nixos/modules/services/matrix/draupnir.nix

```diff
@@ -0,0 +1,272 @@
+      type = lib.types.submodule {
+        options = {
+          enable = lib.mkEnableOption (''
+            pantalaimon, in order to enable E2EE support.
+            If `true`, accessToken is ignored and the username/password below will be
+            used instead. The access token of the bot will be stored in /var/lib/draupnir.
+          '');
+
+          username = lib.mkOption {
+            type = lib.types.str;
+            description = ''
+              Account name on the Matrix homeserver.
+            '';
+          };
+
+          passwordFile = lib.mkOption {
+            type = with lib.types; nullOr path;
+            default = null;
+            description = ''
+              File containing the password for the Matrix account.
+              Make sure this does not contain newlines if writing manually: `:set nofixeol` for vim or -L for nano.
+            '';
+          };
+
+          options = lib.mkOption {
+            type = lib.types.submodule (import ./pantalaimon-options.nix { inherit lib; inherit config; name = "draupnir"; });
+            default = { };
+            description = ''
+              Pass through additional options to the `pantalaimon` service.
+            '';
+          };
+        };
+      };
+    };
+    #endregion
+
+    #region Draupnir settings
+    settings = lib.mkOption {
+      example = lib.literalExpression ''
+        {
+          autojoinOnlyIfManager = true;
+          automaticallyRedactForReasons = [ "spam" "advertising" ];
+        }
+      '';
+      description = ''
+        Draupnir settings (see [Draupnir's default configuration](https://github.com/the-draupnir-project/Draupnir/blob/main/config/default.yaml) for available settings).
+        These settings will override settings made by the module config.
+      '';
+      default = { };
+      type = lib.types.submodule {
+        freeformType = format.type;
+        options = {
+          #region Readonly settings - these settings are not configurable
+          dataPath = lib.mkOption {
+            type = lib.types.str;
+            default = "/var/lib/draupnir";
+            readOnly = true;
+            description = ''
+              The path where Draupnir stores its data.
+
+              ::: {.note}
+                If you want to customize where this data is stored, use a bind mount.
+              :::
+            '';
+          };
+
+          pantalaimon = lib.mkOption {
+            readOnly = true;
+            description = ''
+              `pantalaimon` settings (enables E2E Encryption support).
+              This property is read-only, please configure `services.draupnir.pantalaimon` instead!
+            '';
+            default = {};
+            type = lib.types.submodule {
+              freeformType = format.type;
+              options = {
+                use = lib.mkOption {
+                  type = lib.types.bool;
+                  default = if cfg.pantalaimon.enable then true else false;
```

Comment:
If you access `cfg` or more generally `config` in `default` we require `defaultText`, so it can be rendered in the manual.

Reply:
what would you recommend I put as defaultText? just the string "cfg.pantalaimon.enable"?

Reply:
Yes, and it makes also senseo to reduce the expression in `default` to that.

Reply:
Btw., this was an example and there are likely other options with the same issue.

Reply:
I think I resolved all instances of this? CI still seems to fail though. Ideas?